text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,623 @@
+  infrastructure team.
+  - This process may require one-time or semi-regular manual steps by the
+    infrastructure team, such as registration or renewal of a signing key. Any
+    such manual process must meet the approval of the infrastructure team.
+  - This requirement exists to ensure that Rust builds, including nightly
+    builds, can meet the necessary requirements to allow users to smoothly run
+    the host tools.
+- Providing host tools does not exempt a target from requirements to support
+  cross-compilation if at all possible.
+- All tier 2 requirements apply.
+
+A target may be promoted directly from tier 3 to tier 2 with host tools if it
+meets all the necessary requirements, but doing so may introduce substantial
+additional complexity. If in doubt, the target should qualify for tier 2
+without host tools first.
+
+## Tier 1 target policy
+
+At this tier, the Rust project guarantees that a target builds and passes all
+tests, and will reject patches that fail to build or pass the testsuite on a
+target. We hold tier 1 targets to our highest standard of requirements.
+
+A proposed new tier 1 target must be reviewed and approved by the compiler team
+based on these requirements. In addition, the release team must approve the
+long-term viability of the target, and the additional work of supporting the
+target. For a tier 1 target, this will typically take place via a full RFC
+proposing the target, to be jointly reviewed and approved by the compiler team
+and release team.
+
+In addition, the infrastructure team must approve the integration of the target
+into Continuous Integration (CI), and the tier 1 CI-related requirements. This
+review and approval may take place in a PR adding the target to CI, by an
+infrastructure team member reporting the outcome of a team discussion, or by
+including the infrastructure team in the RFC proposing the target.
+
+- Tier 1 targets must have substantial, widespread interest within the
+  developer community, and must serve the ongoing needs of multiple production
+  users of Rust across multiple organizations or projects. These requirements
+  are subjective, and determined by consensus of the approving teams. A tier 1
+  target may be demoted or removed if it becomes obsolete or no longer meets
+  this requirement.
+- The target must build and pass tests reliably in CI.
+  - The target must not disable an excessive number of tests or pieces of tests
+    in the testsuite in order to do so. This is a subjective requirement.
+- The target must provide as much of the Rust standard library as is feasible
+  and appropriate to provide. For instance, if the target has a standard memory
+  allocator, it must provide an implementation of `alloc` and the associated
+  data structures.
+- Building the target and running the testsuite for the target must not take
+  substantially longer than other targets.
+- If running the testsuite requires additional infrastructure (such as physical
```

Comment:
I believe it is completely fine for a tier 1 target (without host tools support) to have its test suite _built_ on a non-target host and then _only_ run the tests on the target, correct? It would be good to clarify this point as I imagine many targets will not be able to _build_ the test suite in a timely fashion.

Reply:
I believe so, also because a tier 1 target without host tools won't have a native `rustc` to compile the tests.

Reply:
I've documented both of these cases for clarity.